api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -249,21 +249,45 @@ public IActionResult Stream([FromRoute] int? locationId, [FromQuery] string path
             if (dataLocation.Connection == null || dataLocation.Connection?.ConnectionType == ConnectionType.LocalVolume)
             {
                 var safePath = Path.Combine(_storageOptions.GetCapturePath(), path);
-                return GetResult(safePath, path);
+
+                return await GetResult(safePath, path);
             }
             else throw new NotImplementedException($"Location connection type '{dataLocation.Connection?.ConnectionType}' not implemented yet.");
         }
         else
         {
+            _logger.LogInformation("Getting stream for path: {Path}", path);
             var safePath = Path.Combine(_storageOptions.GetUploadPath(), path);
-            return GetResult(safePath, path);
+            return await GetResult(safePath, path);
         }
     }
 
-    private FileStreamResult GetResult(string safePath, string path)
+
+    private async Task<IActionResult> GetResult(string safePath, string path)
     {
-        if (!safePath.FileExists()) throw new NoContentException($"Stream does not exist: '{path}'");
+        _logger.LogInformation("Getting stream for path: {Path}", path);
+        var fileReference = await _fileReferenceService.GetByS3PathAsync(path);
+        _logger.LogInformation("File reference: {FileReference}", fileReference);
+        if (fileReference == null)
+        {
+            return NotFound($"Stream does not exist: '{path}'");
+        }
+
+        if (fileReference.IsSyncedToS3)
```

Comment:
Is there a reason this logic appears to be identical to the code in the FileReferenceService?  It appears like it could be centralized.